core/action/id.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package action
+
+import (
+	coreerrors "github.com/juju/juju/core/errors"
+	"github.com/juju/juju/internal/errors"
+	"github.com/juju/juju/internal/uuid"
+)
+
+// UUID represents an action unique identifier.
+type UUID string
```

Comment:
Issue: remove this, a specific UUID type is not needed.

Per Joe "Please don't add new UUID types under core packages."

Use `uuid.UUID` and string instead.